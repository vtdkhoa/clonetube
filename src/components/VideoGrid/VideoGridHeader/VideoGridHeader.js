import React from 'react'
import PropTypes from 'prop-types'
import './VideoGridHeader.scss'

const VideoGridHeader = props => {
  return (
    <div className="video-grid-header">
      <span className="title">{props.title}</span>
    </div>
  )
}

VideoGridHeader.propTypes = {
  title: PropTypes.string
}

export default VideoGridHeader