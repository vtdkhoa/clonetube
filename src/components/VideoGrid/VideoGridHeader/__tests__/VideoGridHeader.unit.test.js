import React from 'react'
import VideoGridHeader from '../VideoGridHeader'
import { shallow } from 'enzyme'

describe('VideoGridHeader', () => {
  test('renders without props', () => {
    const wrapper = shallow(<VideoGridHeader/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with empty title', () => {
    const wrapper = shallow(
      <VideoGridHeader title=""/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with title', () => {
    const wrapper = shallow(
      <VideoGridHeader title="Sample title"/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})