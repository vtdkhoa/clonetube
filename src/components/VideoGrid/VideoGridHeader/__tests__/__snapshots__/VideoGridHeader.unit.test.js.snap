// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGridHeader renders with empty title 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;

exports[`VideoGridHeader renders with title 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  >
    Sample title
  </span>
</div>
`;

exports[`VideoGridHeader renders without props 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;
